fix(gasket-module): append upstream src to SRC_URI instead of overwriting

fmoessbauer opened the pull request:
Fixes the regression introduced in 8bc2be36d6 on clean builds.
The upstream src has to be appended to SRC_URI to not overwrite
the referenced local files.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
fix(gasket-module): append upstream src to SRC_URI instead of overwriting

Fixes the regression introduced in 8bc2be36d6 on clean builds.
The upstream src has to be appended to SRC_URI to not overwrite
the referenced local files.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
ci: build gasket module for upstream debian kernel

This patch adds a meta target to test the build of all
packages. This includes a build of the gasket module for
the stock debian upstream kernel.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer commented:
@chombourger I just added a test to also build the gasket module in the ci. By that we should be safe from regressions like the mentioned one.

jan-kiszka commented on kas/meta-coral.yml:
```diff
@@ -41,3 +42,5 @@ local_conf_header:
   ccache: |
     USE_CCACHE = "1"
+  kernel: |
+    KERNEL_NAME = "${DISTRO_ARCH}"
```
Only works for amd64 and arm64...

fmoessbauer replied:
Is that a problem here? This file will only be used locally for testing and in case we add support for further architectures, we still can overwrite the kernel entry.

jan-kiszka replied:
Just wanted to point this out, didn't look at the context in which this is used.

jan-kiszka commented on kas/meta-coral.yml:
```diff
@@ -33,6 +33,7 @@ target:
   - python-gast
   - bazel-bootstrap
   - gasket-dkms
+  - gasket-module-${DISTRO_ARCH}
```
KERNEL_NAME

fmoessbauer committed:
fix: disable cross-compilation of gasket-module on distro kernels

The gasket-module cannot be cross compiled for the arm64 distro
kernel due to incompatible host vs target build dependencies of
the kernel headers (cpp-10 package).

Without sbuilder we do not build in a clean environment, hence
we have to disable cross-compilation to not attract incompatible
deps.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer committed:
ci: Increase free diskspace on builder

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

fmoessbauer commented:
@chombourger , @jan-kiszka : Please review again.

jan-kiszka commented on recipes-test/ci-all-packages/ci-all-packages.bb:
```diff
@@ -0,0 +1,28 @@
+
+# This is a meta recipe to test the building of all packages
+
+inherit dpkg-raw
```
Hmm, do we actually need to build a package for this? As long as we do not build an image and install this as meta package, we may even skip this.

fmoessbauer replied:
Without inheriting something (?), the `DEPENDS` are not picked up.
This package will only end up in the local build tree, which is used by nobody.

But maybe we should rename the recipe to `meta-coral-ci` to avoid name clashes with other layers that include meta-coral.

jan-kiszka replied:
Yeah, would likely still need the base class... Anyway, not that important.

jan-kiszka commented on recipes-test/ci-all-packages/ci-all-packages.bb:
```diff
@@ -0,0 +1,28 @@
+    gasket-module-${KERNEL_NAME} \
+    tensorflow                   \
+    pycoral                      \
+"
```
You never add a proper newline at EOF - not a problem per se, just a bit ugly.

fmoessbauer replied:
Indeed. Can be fixed on next version.

fmoessbauer commented:
If there are no objections, I'll merge this tonight.

chombourger commented:
> If there are no objections, I'll merge this tonight.

LGTM
